File: hello_world/hello_world.c
#include <stdio.h>
 
int main()
{
  char s1[] = "HELLO WORLD";
  char s2[] = {'H','e','l','l','o',' ','w','o','r','l','d','\0'};
 
  printf("%s %s", s1, s2);
 
  return 0;
}

File: hello_world/python/helloworld.py

def main():
    import os
    import re
    import subprocess

    pattern = re.compile("[Hh][Ee][Ll][Ll][Oo]_*[wW][Oo][Rr][Ll][Dd]+\.py")
    a = os.listdir(".")
    for i in  a:
        if pattern.match(i) and  i != str(os.path.basename(__file__)):
            subprocess.call(["python", i])
            break
        else:
            print "Hello World!"
            exit(0)

if __name__ == "__main__":
    main()
